db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
I mean what you mentioned in the first paragraph. The second paragraph I mean it overlaps with one file so although trivial move is not enabled, we only need to rewrite minimum data to get the file to L1.